CMakeLists.txt

```diff
@@ -43,6 +43,10 @@ option(DEVILUTIONX_SYSTEM_LIBSODIUM "Use system-provided libsodium" ON)
 cmake_dependent_option(DEVILUTIONX_STATIC_LIBSODIUM "Link static libsodium" OFF
                        "DEVILUTIONX_SYSTEM_LIBSODIUM AND NOT DIST" ON)
 
+option(DEVILUTIONX_SYSTEM_LIBFMT "Use system-provided libfmt" OFF)

```

Comment:
This should probably default to `ON`

* Provided by `libfmt-dev` on Ubuntu and Debian
* Provided by `fmt` on Mac (`brew install fmt`)

Reply:
The issue is that the `libfmt-dev` package is [quite old](https://packages.ubuntu.com/search?keywords=libfmt-dev) on Ubuntu 18.04 (4.0.0). The latest version is 7.1.3. There have been some breaking changes since then. Ubuntu 18.04 is still supported until April 2023.

What we could do is check the version that is available on the system, and if its < 7.0.0 then download `fmt` using CMake?

Reply:
> What we could do is check the version that is available on the system, and if its < 7.0.0 then download fmt using CMake?

This sounds good. This can be done by replacing `find_package(fmt REQUIRED)` with `find_package(fmt 7.0.0)` and checking `fmt_FOUND`.